src/Essentials/src/AppInfo/AppInfo.android.cs

```diff
@@ -46,14 +47,13 @@ public void ShowSettingsUI()
 			context.StartActivity(settingsIntent);
 		}
 
-		static AppTheme GetRequestedTheme() => (Application.Context.Resources.Configuration.UiMode & UiMode.NightMask) switch
-		{
-			UiMode.NightYes => AppTheme.Dark,
-			UiMode.NightNo => AppTheme.Light,
-			_ => AppTheme.Unspecified
-		};
-
-		public AppTheme RequestedTheme => _requestedTheme.Value;
+		public AppTheme RequestedTheme
+			=> (Application.Context.Resources.Configuration.UiMode & UiMode.NightMask) switch
```

Comment:
@jonathanpeppers or @PureWeen is there somewhere we have the Context cached already for the app that we can use here? 

Or do we need to look for the theme change on the platform level somehow and update the lazy loaded value instead?  

It seems like asking for this dynamically is going to undo some work around performance given it crosses the interop boundary.

Reply:
This is an idea on how we could do it:

https://ondrejbalas.com/an-implementation-of-lazy-with-invalidation/

But maybe we can find something else that is built-in.

Reply:
Hmm, it's not obvious to me how we'd invalidate this. In MAUI we trigger theme changes by overriding `OnConfigurationChanged` on the `Activity`. Is there a listener/event equivalent to this that we can watch at the `Essentials` level?  We are using `IActivityLifecycleCallbacks` inside `Essentials` but that doesn't appear to give us any info about `OnConfigurationChanged`. It seems like you can only be aware of an invalidate via an override on the `Activity`? 

This feels like something we'd possibly solve higher up instead of this low down.  `Essentials` is dumb for a reason, and then at the `Controls` or `Core` level we'd cache this value and then invalidate that cached value once `OnConfigurationChanged` has fired.

Unless there's an obvious way to invalidate this in `Essentials` that I'm not seeing?

> I was seeing every View calls into this API

Can we make every View call into a different API that's able to cache/invalidate this?

Reply:
I think the path here is that the MainActivity needs to invalidate the conifugration as that is where it triggers the change: https://developer.android.com/develop/ui/views/theming/darktheme itis a shame that IActivityLifecycleCallbacks doesn't give it to us, seems like an oversight from Google IMHO. 

It looks like they have a listener we could loop into https://developer.android.com/reference/androidx/core/content/OnConfigurationChangedProvider

This is the route most likely IMHO

Reply:
I think I have taken inspiration from @PureWeen's comment. We already have all the bits in place in Controls to handle configuration changes - and that is where all the use of the values are.